examples/basic-betting-market/index.rsh

```diff
@@ -96,70 +80,69 @@ export const main = Reach.App(() => {
       return [false, trueBetters, truePool, falseBetters, falsePool];
     });
 
-  BP.phase(Phase.AwaitingResult());
-  commit();
-
   assert(truePool + falsePool == balance());
 
   // Await result
-  // const result = true;
+  BP.phase(Phase.AwaitingResult());
+  commit();
+  const result = true;
 
   // Pay winners
-  const numWinners = /* result ? */ trueBetters /* : falseBetters */;
-  const winnerBets = /* result ? */ trueBets /* : falseBets */;
-  const winnerPool = /* result ? */ truePool /* : falsePool */;
-  const loserPool = /* result ? */ falsePool /* : truePool */;
-
-  // Do truncation in favor of winners, against bookie
-  const organizerCut = 0; //fx2uint(fxmul(bookieCut, uint2fx(loserPool)));
-  const initialPayoutPool = loserPool - organizerCut;
+  awaitOrganizer(OA.startPayout);
 
-  const checkWinner = p => {
-    const bet = winnerBets[p];
-    check(bet != Maybe(UInt).None());
-    check(bet != Maybe(UInt).Some(0));
-  }
   const unwrap = m => {
     switch (m) {
-      case Some: return m;
+      case Some:
+        return m;
+      case None:
+        assert(false, "unwrap on None");
+        return [false, 0];
+    }
+  }
+  const checkWinner = p => {
+    const bet = bets[p];
+    switch (bet) {
+      case Some:
+        check(bet[0] == result);
+        check(bet[1] > 0);
       case None:
-        assert(false, "unwrap None");
-        return 0;
+        check(false);
     }
   }
 
-  awaitOrganizer(OA.startPayout);
-  const [keepGoing_, unpaidWinners, payoutPool] =
-    parallelReduce([true, numWinners, initialPayoutPool])
+  const [numWinners, winnerPool, loserPool] = /*result ?*/ [trueBetters, truePool, falsePool]
```

Comment:
I don't think it is useful to test and try to avoid the verification errors on something you know you don't really want. I think you should use dynamic control-flow to get some static data-flow and I think this might solve the problem with tracking the winning pool